wazimap_ng/profile/admin/admins/profile_key_metrics_admin.py

```diff
@@ -37,16 +42,18 @@ class ProfileKeyMetricsAdmin(SortableAdminMixin, BaseAdminModel, HistoryAdmin):
 
     help_texts = ["denominator", ]
 
-    fields = ["profile", "variable", "subindicator", "subcategory", "denominator", "label"]
-    search_fields = ("label", )
+    search_fields = ("label",)
 
+    def get_readonly_fields(self, request, obj=None):
+        if obj:  # editing an existing object
+            return ("profile",) + self.readonly_fields
+        return self.readonly_fields
```

Comment:
The codeclimate warning about repeating code might actually be correct here. But it can stay like this for now